u1-request-response/app.js

```diff
@@ -0,0 +1,149 @@
+// Import libraries
+const express = require('express')
+
+// Setup app
+const app = express()
+
+// Middleware
+app.use(express.static('public'))
+
+// Routes
+// Localhost:3000
+app.get('/',(req, res) => {
+  res.send('better world')
+})
+
+// Localhost:3000/html
+app.get('/html', (req, res) => {
+  res.send(`
+    <html><body>
+      <h1>Technologies</h1>
+      <ul>
+        <li>JavaScript</li>
+        <li>Node</li>
+        <li>Express</li>
+      </ul>
+    </body></html>
+  `)
+})
+
+// Localhost:3000/json
+app.get('/json', (req, res) => {
+  res.json({ 
+    technologies: [
+      'javascript', 'typescript', 'Node', 'Express'
+    ]
+  })
+})
+
+// localhost:3000/laps
+app.get('/laps', (req, res) => {
+  res.send(`
+<html><body>
+  <li>1. <a href="/profiles/wanpen">/profiles/wanpen</a></li>
+  <li>2. <a href="/api/profiles/wanpen">/api/profiles/wanpen</a></li>
+  <li>3. <a href="/cat-simulator">/cat-simulator</a></li>
+  <li>4. <a href="/admin/grades">/admin/grades</a> AND <a href="/api/grades">/api/grades</a> AND <a href="/boredom">/boredom</a></li>
+  </body></html>
+  `)
+})
+
+// localhost:3000/profiles/wanpen
+app.get('/profiles/wanpen', (req, res) => {
+  res.send(`
+
+  <h1>Wanpen Phunma</h1>
+  <h3>nickname: Oil</h3>
+  <h3>likes: Kpop, Movies, Listen to music</h3>
+  <h3>motto: Where there's a will, there's a way</h3>
+
+ `)
+})
+
+// localhost:3000/api/profiles/wanpen
+app.get('/api/profiles/wanpen', (req, res) => {
+  res.send(`
+{
+  "name" : "Wanpen Phunma",
+  "nickname": "Oil",
+  "likes": [
+    "Kpop", 
+    "Movies", 
+    "Listen to music",
+  ],
+  "motto": Where there's a will, there's a way"
+}
+ `)
+})
+
+// Localhost:3000/cat-simulator
+app.get('/cat-simulator', (req, res) => {
+  res.send(`
+  <html>
+  <head><link rel="stylesheet" href="/styles.css"></head>
+  <body class="cats">
+  <a href='http://localhost:3000/cat-result'> Pet the cat </a><br><br>
```

Comment:
URL นี้จะใช้ได้เฉพาะในเครื่องเท่านั้น คุณต้องใช้ URL "base-relative" เช่น `href="/cat-result"`
This URL will only work locally. You need to use a "base-relative" URL, like this: `href="/cat-result"`.